Branch 3.x (3.2.x) fails assert

### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

I expect that building on the 3.x branch succeeds.

### Current behavior

The build fails a static assert
https://github.com/eProsima/Fast-DDS/blob/95ea270a00c554f080502ead20620bae014758cc/src/cpp/rtps/messages/MessageReceiver.cpp#L870

The full error
```
In file included from /work/fastdds/include/fastdds/fastdds_dll.hpp:23,
                 from /work/fastdds/include/fastdds/rtps/common/EntityId_t.hpp:22,
                 from /work/fastdds/src/cpp/rtps/messages/MessageReceiver.cpp:24:
/work/fastdds/src/cpp/rtps/messages/MessageReceiver.cpp: In member function ‘bool eprosima::fastdds::rtps::MessageReceiver::proc_Submsg_Data(eprosima::fastdds::rtps::CDRMessage_t*, eprosima::fastdds::rtps::SubmessageHeader_t*, eprosima::fastdds::rtps::EntityId_t&, bool) const’:
/ros_work/fastdds-spy/build/fastdds/include/fastdds/config.hpp:123:51: error: static assertion failed: TODO before version 3.2 : "Pass keyFlag in serializedPayload, and always pass input data upwards"
  123 |     static_assert((FASTDDS_VERSION_MAJOR < major) ||                                   \
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  124 |             (FASTDDS_VERSION_MAJOR == major && FASTDDS_VERSION_MINOR < minor),  \
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/work/fastdds/src/cpp/rtps/messages/MessageReceiver.cpp:870:13: note: in expansion of macro ‘FASTDDS_TODO_BEFORE’
  870 |             FASTDDS_TODO_BEFORE(3, 2, "Pass keyFlag in serializedPayload, and always pass input data upwards");
      |             ^~~~~~~~~~~~~~~~~~~
gmake[2]: *** [src/cpp/CMakeFiles/fastdds.dir/build.make:1924: src/cpp/CMakeFiles/fastdds.dir/rtps/messages/MessageReceiver.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
gmake[1]: *** [CMakeFiles/Makefile2:976: src/cpp/CMakeFiles/fastdds.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2
```

### Steps to reproduce

Clone the repository with the 3.x branch.  
colcon build

### Fast DDS version/commit

Branch 3.x

### Platform/Architecture

Other. Please specify in Additional context section.

### Transport layer

Default configuration, UDPv4 & SHM

### Additional context

Building in an Ubuntu 24.04 docker container (docker.io/library/ubuntu:noble)

### XML configuration file

_No response_

### Relevant log output

_No response_

### Network traffic capture

_No response_

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -29,7 +29,7 @@ endif()
 ###############################################################################
 # Project                                                                     #
 ###############################################################################
-project(fastdds VERSION "3.2.0" LANGUAGES C CXX)
+project(fastdds VERSION "3.1.0" LANGUAGES C CXX)
 
 set(PROJECT_NAME_LARGE "Fast DDS")
 string(TOUPPER "${PROJECT_NAME}" PROJECT_NAME_UPPER)

--- a/package.xml
+++ b/package.xml
@@ -2,7 +2,7 @@
 <?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
 <package format="3">
   <name>fastdds</name>
-  <version>3.2.0</version>
+  <version>3.1.0</version>
   <description>
      *eprosima Fast DDS* is a C++ implementation of the DDS (Data Distribution Service) standard of the OMG (Object Management Group). eProsima Fast DDS implements the RTPS (Real Time Publish Subscribe) protocol, which provides publisher-subscriber communications over unreliable transports such as UDP, as defined and maintained by the Object Management Group (OMG) consortium. RTPS is also the wire interoperability protocol defined for the Data Distribution Service (DDS) standard. *eProsima Fast DDS* expose an API to access directly the RTPS protocol, giving the user full access to the protocol internals.
   </description>

